### PR TITLE
unit testing

### PR DESCRIPTION
**Pull request by larshp**
* split test method `get_extrema()` into multiple test cases
* split testcase `call_reader_and_decode`
* more unit tests added
* constant `c_indent_number_characters` changed to CONSTANT in `zcl_aff_writer`

### CODE REVIEW

**Comment by schneidermic0** [comment]
> Generating INTF AFF json schema file now working 🎉 run npm install && npm run aff
Nice. 

I'll try to review the changes as soon as possible. Be patient with me, please

**Review by schneidermic0 — COMMENTED** [review]
Thanks, @larshp. I have just added a small comment/idea.

Since this is a bigger change (including new tests which seem to be on the exception list). I would like to run the unit tests once more in an ABAP system. Unfortunately, I don't have access to a system, today :(

File: src/zcl_aff_writer.clas.testclasses.abap [hunk 1]
```diff
@@ -274,39 +280,67 @@ CLASS ltcl_type_writer IMPLEMENTATION.
...
     DATA(element_name) = `CATEGORY`.
     DATA(abap_doc_act) = cut->call_reader_and_decode( name_of_source = name_of_source element_name = element_name ).
     DATA abap_doc_exp TYPE zcl_aff_abap_doc_parser=>abap_doc.
     abap_doc_exp = VALUE #( description = `This is an enum` title = `myCategory` enumvalues_link = `zcl_aff_test_types.data:enum_values` ).
     cl_abap_unit_assert=>assert_equals( exp = abap_doc_exp act = abap_doc_act ).
+  ENDMETHOD.
 
-    element_name = `AFF_TEST_TYPE-INNER_STRUC`.
-    abap_doc_act = cut->call_reader_and_decode( name_of_source = name_of_source element_name = element_name ).
+  METHOD call_reader_and_decode2.
```
[the remark this answers is not in the document]
Comment (larshp) [reply src/zcl_aff_writer.clas.testclasses.abap:292]: suggest doing that in a follow up PR

**Review by schneidermic0 — APPROVED** [review]
Unit tests are also green in the ABAP system 👍

**Pull request merged**